cuebot/src/main/java/com/imageworks/spcue/PrometheusMetricsCollector.java

```diff
@@ -119,6 +123,52 @@ public class PrometheusMetricsCollector {
             .labelNames("env", "cuebot_host", "render_node", "job_name", "frame_name", "frame_id")
             .register();
 
+    // Monitoring event metrics
+    private static final Counter monitoringEventsPublishedCounter =
+            Counter.build().name("cue_monitoring_events_published_total")
+                    .help("Total number of monitoring events published to Kafka")
+                    .labelNames("env", "cuebot_host", "event_type").register();
+
+    private static final Counter monitoringEventsDroppedCounter =
+            Counter.build().name("cue_monitoring_events_dropped_total")
+                    .help("Total number of monitoring events dropped due to queue overflow")
+                    .labelNames("env", "cuebot_host", "event_type").register();
+
+    private static final Gauge monitoringEventQueueSize =
+            Gauge.build().name("cue_monitoring_event_queue_size")
+                    .help("Current size of the monitoring event publishing queue")
+                    .labelNames("env", "cuebot_host").register();
+
+    private static final Gauge elasticsearchIndexQueueSize =
+            Gauge.build().name("cue_elasticsearch_index_queue_size")
+                    .help("Current size of the Elasticsearch indexing queue")
+                    .labelNames("env", "cuebot_host").register();
+
+    private static final Counter frameCompletedCounter = Counter.build()
+            .name("cue_frames_completed_total").help("Total number of frames completed")
+            .labelNames("env", "cuebot_host", "state", "show").register();
+
+    private static final Counter jobCompletedCounter =
+            Counter.build().name("cue_jobs_completed_total").help("Total number of jobs completed")
+                    .labelNames("env", "cuebot_host", "state", "show").register();
+
+    private static final Histogram frameRuntimeHistogram = Histogram.build()
+            .name("cue_frame_runtime_seconds").help("Histogram of frame runtimes in seconds")
+            .labelNames("env", "cuebot_host", "show", "layer_type")
+            .buckets(60, 300, 600, 1800, 3600, 7200, 14400, 28800, 86400).register();
+
+    private static final Histogram frameMemoryHistogram = Histogram.build()
+            .name("cue_frame_memory_bytes").help("Histogram of frame peak memory usage in bytes")
+            .labelNames("env", "cuebot_host", "show", "layer_type")
+            .buckets(256L * 1024 * 1024, 512L * 1024 * 1024, 1024L * 1024 * 1024,
+                    2048L * 1024 * 1024, 4096L * 1024 * 1024, 8192L * 1024 * 1024,
+                    16384L * 1024 * 1024, 32768L * 1024 * 1024)
+            .register();
```

Comment:
Having prometheus metrics per frame might be too costly. I suggest changing this to collect the max per layer as both runtime and memory is supposed to be very similar between frames of the same layer.

layerMaxRuntimeHistogram
layerMaxMemoryHistogram